client/ayon_deadline/lib.py

```diff
@@ -43,3 +48,368 @@ def get_instance_job_envs(instance) -> "dict[str, str]":
+    JobDelay: Optional[str] = field(default=None)  # <dd:hh:mm:ss>
+    Scheduled: Optional[str] = field(
+        default=None)  # <Day of the Week><Start/Stop>Time=<HH:mm:ss>
+
+    # Scripts
+    PreJobScript: Optional[str] = field(default=None)  # Default: blank
+    PostJobScript: Optional[str] = field(default=None)  # Default: blank
+    PreTaskScript: Optional[str] = field(default=None)  # Default: blank
+    PostTaskScript: Optional[str] = field(default=None)  # Default: blank
+
+    # Event Opt-Ins
+    EventOptIns: Optional[str] = field(
+        default=None)  # Default blank (comma-separated list)
+
+    # Environment
+    EnvironmentKeyValue: Any = field(
+        default_factory=partial(DeadlineKeyValueVar, "EnvironmentKeyValue"))
+    IncludeEnvironment: Optional[bool] = field(default=False)  # Default: false
+    UseJobEnvironmentOnly: Optional[bool] = field(default=False)  # Default: false
+    CustomPluginDirectory: Optional[str] = field(default=None)  # Default blank
+
+    # Job Extra Info
+    ExtraInfo: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "ExtraInfo"))
+    ExtraInfoKeyValue: Any = field(
+        default_factory=partial(DeadlineKeyValueVar, "ExtraInfoKeyValue"))
+
+    OverrideTaskExtraInfoNames: Optional[bool] = field(
+        default=False)  # Default false
+
+    TaskExtraInfoName: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "TaskExtraInfoName"))
+
+    OutputFilename: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputFilename"))
+    OutputFilenameTile: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputFilename{}Tile"))
+    OutputDirectory: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputDirectory"))
+
+    AssetDependency: str = field(
+        default_factory=partial(DeadlineIndexedVar, "AssetDependency"))
+
+    TileJob: bool = field(default=False)
+    TileJobFrame: int = field(default=0)
+    TileJobTilesInX: int = field(default=0)
+    TileJobTilesInY: int = field(default=0)
+    TileJobTileCount: int = field(default=0)
+
+    MaintenanceJob: bool = field(default=False)
+    MaintenanceJobStartFrame: int = field(default=0)
+    MaintenanceJobEndFrame: int = field(default=0)
+
+
+@dataclass
+class AYONDeadlineJobInfo(DeadlineJobInfo):
+    """Contains additional AYON variables from Settings for internal logic."""
+
+    # AYON custom fields used for Settings
+    UsePublished: Optional[bool] = field(default=None)
+    UseAssetDependencies: Optional[bool] = field(default=None)
+    UseWorkfileDependency: Optional[bool] = field(default=None)
+
+    MachineList: Optional[str] = field(
+        default=None)  # Default blank (comma-separated list)
+    MachineListDeny: Optional[bool] = field(default=None)
+
+    AdditionalJobInfo: Optional[str] = field(default=None)  # Default: blank
+    AdditionalPluginInfo: Optional[str] = field(default=None)  # Default: blank
```

Comment:
What's this? How would one specify additional job info or plug-in info? Doesn't seem like something Deadline would support?

Reply:
This is for all not explicitly implemented fields, as we are not implementing them all. For example there is nothing in code/Settings to control `ArchiveOnComplete` in `JobInfo`, with this values in Settings you could control it:
<img width="512" alt="chrome_ugIHMDvWdA" src="https://github.com/user-attachments/assets/e46b3245-daec-4384-a829-c7a9140dba71">